packages/@sanity/default-layout/src/navbar/Navbar.tsx

```diff
@@ -1,178 +1,328 @@
+
+  const rootState = useMemo(
+    () => (HAS_SPACES && router?.state?.space ? {space: router?.state?.space} : {}),
+    [router]
+  )
+
+  const shouldRender = {
+    brandingCenter: mediaIndex <= 1,
+    collapsedPresenceMenu: mediaIndex <= 1,
+    hints: mediaIndex > 1 && sidecar && sidecar.isSidecarEnabled && sidecar.isSidecarEnabled(),
+    loginStatus: mediaIndex > 1,
+    searchFullscreen: mediaIndex <= 1,
+    spaces: HAS_SPACES && mediaIndex >= 3,
+    statusContainer: mediaIndex > 1,
+    tools: mediaIndex >= 3,
+  }
+
+  useGlobalKeyDown((e) => {
+    if (e.key === 'Escape' && searchOpen) {
+      setSearchOpen(false)
+      searchButtonElement?.focus()
+    }
+  })
+
+  const handleToggleSearchOpen = useCallback(() => {
+    setSearchOpen((prev) => {
+      if (prev) {
+        searchButtonElement?.focus()
+      }
+
+      return !prev
+    })
+  }, [searchButtonElement])
+
+  useEffect(() => {
+    if (searchIsOpen && !shouldRender.searchFullscreen) {
+      setSearchOpen(false)
+      searchIsOpen(false)
+    }
+  }, [searchIsOpen, shouldRender.searchFullscreen])
+
+  useEffect(() => {
+    searchIsOpen(searchOpen)
+
+    if (searchOpen) {
+      inputElement?.focus()
+    }
+  }, [inputElement, searchButtonElement, searchIsOpen, searchOpen])
+
+  const LinkComponent = useCallback(
+    (linkProps) => {
+      return <StateLink state={rootState} {...linkProps} />
+    },
+    [rootState]
+  )
 
   return (
-    <Card className={className} data-search-open={searchIsOpen} scheme="dark">
-      <div className={styles.hamburger}>
-        <Button
-          aria-label="Open menu"
-          icon={HamburgerIcon}
-          kind="simple"
-          onClick={onToggleMenu}
-          padding="small"
-          title="Open menu"
-          tone="navbar"
-        />
-      </div>
-      <div className={styles.branding}>
-        <StateLink state={rootState} className={styles.brandingLink}>
-          <Branding projectName={config && config.project.name} />
-        </StateLink>
-      </div>
-      {HAS_SPACES && (
-        <div className={styles.datasetSelect}>
-          <DatasetSelect isVisible={showToolMenu} tone="navbar" />
-        </div>
-      )}
-      <div className={styles.createButton}>
-        <LegacyLayerProvider zOffset="navbarPopover">
-          <Tooltip
-            content={
-              createAnyPermission.granted ? (
-                <span className={styles.createButtonTooltipContent}>Create new document</span>
-              ) : (
-                <Card padding={2} radius={1}>
-                  <InsufficientPermissionsMessage
-                    currentUser={currentUser}
-                    operationLabel="create any document"
-                  />
-                </Card>
-              )
-            }
-          >
-            <div>
+    <Root
+      paddingY={2}
+      paddingLeft={2}
+      paddingRight={3}
+      scheme="dark"
+      display="flex"
+      sizing="border"
+    >
```

Comment:
This can be simplified to: `<Root padding={2} scheme="dark">`

NOTE: the right padding is currently wrong. It should be `2`, and not `3`.